sdk/keyvault/keyvault-admin/test/public/accessControlClient.spec.ts

```diff
@@ -0,0 +1,107 @@
+    await recorder.stop();
+  });
+
+  // The tests follow
+
+  it("listRoleDefinitions", async function() {
+    const expectedType = "Microsoft.Authorization/roleDefinitions";
+    let receivedRoles: string[] = [];
+
+    for await (const roleDefinition of client.listRoleDefinitions(globalScope)) {
+      // Each role definition will have the shape of:
+      //
+      //   {
+      //     id: 'Microsoft.KeyVault/providers/Microsoft.Authorization/roleDefinitions/<ID>',
+      //     name: '<ID>',
+      //     type: '<role-type>',
+      //     roleName: '<role-name>',
+      //     // ...
+      //   }
+      //
+      assert.equal(roleDefinition.type, expectedType);
+      receivedRoles.push(roleDefinition.roleName!);
+    }
+
+    // Roles might change
+    assert.ok(receivedRoles.length);
+  });
+
+  it("listRoleAssignments", async function() {
+    const expectedType = "Microsoft.Authorization/roleAssignments";
+    let receivedRoles: string[] = [];
+
+    for await (const roleAssignment of client.listRoleAssignments(globalScope)) {
+      // Each role assignment will have the shape of:
+      //
+      //   {
+      //     id: '/providers/Microsoft.Authorization/roleAssignments/<ID>',
+      //     name: '<ID>',
+      //     type: '<role-type>',
+      //     // ...
+      //   }
+      //
+      assert.equal(roleAssignment.type, expectedType);
+      receivedRoles.push(roleAssignment.name);
+    }
+
+    // Roles might change
+    assert.ok(receivedRoles.length);
+  });
+
+  it("createRoleAssignment, getRoleAssignment and deleteRoleAssignment", async function() {
+    // First, deleting any existing assignment, just in case.
+    for await (const roleAssignment of client.listRoleAssignments(globalScope)) {
+      // Removing all roles from this object ID might kick us out of the system.
+      // IMPORTANT: Make sure CLIENT_OBJECT_ID isn't the Object ID of the principal used to authenticate.
+      if (roleAssignment.properties.principalId === env.CLIENT_OBJECT_ID) {
+        await client.deleteRoleAssignment(globalScope, roleAssignment.name);
+      }
+    }
+
+    const roleDefinition = (await client.listRoleDefinitions(globalScope).next()).value;
+    const name = generateFakeUUID();
+
+    let assignment = await client.createRoleAssignment(
+      globalScope,
+      name,
+      roleDefinition.id!,
+      env.CLIENT_OBJECT_ID
+    );
+    assert.equal(assignment.name, name);
+    assert.equal(assignment.properties?.roleDefinitionId, roleDefinition.id);
+    assert.equal(assignment.properties?.principalId, env.CLIENT_OBJECT_ID);
+
+    assignment = await client.getRoleAssignment(globalScope, name);
+    assert.equal(assignment.name, name);
+    assert.equal(assignment.properties?.roleDefinitionId, roleDefinition.id);
+    assert.equal(assignment.properties?.principalId, env.CLIENT_OBJECT_ID);
+
+    assignment = await client.deleteRoleAssignment(globalScope, name);
+    assert.equal(assignment.name, name);
```

Comment:
forgive me as I don't know the details of this lib, should we/is there a way to verify that the roll assignment has been removed?

Reply:
Depending on the role, I would need to use the CLIENT_OBJECT_ID as the principal to authenticate a KeyVault client and catch the error it generates.

If it helps, I locked myself out of the system since I first used the same credential to authenticate the client as to assign and remove roles, so I ended up removing all my credentials through the test.

Reply:
I'd rather avoid doing that setup today, plus that would be more a integration test, since it's about whether the service ended up doing what it reported to have done, if that makes sense.

Reply:
`getRollAssignment()` would probably fail after deletion? maybe catch then verify that error?

Reply:
Can do!

Reply:
Done!